src/etc/inc/dyndns.class

```diff
@@ -998,7 +998,7 @@
 						$output->domain_records = array();
 					}
 					foreach($output->domain_records as $dnsRecord) {
-						if ($this->_dnsHost == $dnsRecord->name) {
+						if ($this->_dnsHost == $dnsRecord->name && $dnsRecord->type == 'A') {
```

Comment:
When trying to update the `@` record, without checking type here, as written would try to update the NS records instead.